js/components/map.js

```diff
@@ -261,24 +265,27 @@ Fliplet.FormBuilder.field('map', {
 
         if (val.address && val.address.label) {
           val = {
-            address: val.address.label
+            address: val.address.label,
+            latLong: null
           };
         }
 
-        if (val.address && !this.suggestionSelected && !this.lastChosenAutocompleteValue && !this.readonly) {
-          if (val.latLong) {
-            this.mapAddressField.set(val.address);
-            this.mapField.set(val.address);
-          } else {
-            this.mapAddressField.set(val.address);
-          }
+        if (val.address && !this.suggestionSelected && !this.lastChosenAutocompleteValue) {
+          setTimeout(() => {
+            if (val.latLong) {
+              this.mapAddressField.set(val.address);
+              this.mapField.set(val.address);
+            } else {
+              this.mapAddressField.set(val.address);
+            }
 
-          this.suggestionSelected = true;
-          this.lastChosenAutocompleteValue = val.address;
-          this.addressSuggestions = [];
-          this.isSelectOnMapClicked = true;
+            this.suggestionSelected = true;
+            this.lastChosenAutocompleteValue = val.address;
+            this.addressSuggestions = [];
+            this.isSelectOnMapClicked = true;
 
-          return;
+            return;
```

Comment:
@armine-fliplet the `return` here is pointless, as it just exits from the setTimeout callback, so it does nothing at the end of function.